presto-main/src/main/java/com/facebook/presto/metadata/StaticCatalogStore.java

```diff
@@ -36,25 +39,31 @@
 public class StaticCatalogStore
 {
     private static final Logger log = Logger.get(StaticCatalogStore.class);
+    public static final String SECRETS_MANAGER_ENABLED = "secretsManager.enabled";
+    public static final String CONNECTOR_NAME = "connector.name";
+    private static final Map<String, Map<String, String>> secrets = new ConcurrentHashMap<>();
```

Comment:
It looks like we load this in one shot.  I'd recommend using an immutable map to store this, and we can wrap it in an `AtomicReference` to valid it's set once and only once.